airflow/providers/google/cloud/operators/bigquery_reservation.py

```diff
@@ -0,0 +1,150 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""This module contains Google BigQuery Reservation Service operators."""
+from __future__ import annotations
+
+from typing import Any, Sequence
+
+from airflow.models import BaseOperator
+from airflow.providers.google.cloud.hooks.bigquery_reservation import BigQueryReservationServiceHook
+
+
+class BigQueryBiEngineReservationCreateOperator(BaseOperator):
+    """
+    Create or Update BI engine reservation.
+
+    :param location: The BI engine reservation location. (templated)
+    :param size: The BI Engine reservation memory size (GB). (templated)
+    :param project_id: (Optional) The name of the project where the reservation
+        will be attached from. (templated)
+    :param gcp_conn_id: (Optional) The connection ID used to connect to Google Cloud. (templated)
+    :param delegate_to: Account to impersonate using domain-wide delegation of authority,
```

Comment:
since we have deprecated `delegate_to` in google provider so can avoid adding it here?